### PR TITLE
[GH-380] - Add User Refresh Token Management Functionality to SDK

### PR DESCRIPTION
**Pull request by developerkunal**
### 🔧 Changes
- Add `GetRefreshTokens` function to retrieve details for a user's refresh tokens.
- Add `DeleteRefreshTokens` function to delete all refresh tokens for a user.
- Update Go version from `1.21.9` to `1.21` to ensure compatibility with [govulncheck-action](https://github.com/golang/govulncheck-action)  and utilize the automatic version detection feature.

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: management/user.go [hunk 1]
```diff
@@ -724,3 +749,22 @@ func (m *UserManager) DeleteAllAuthenticationMethods(ctx context.Context, userID
...
 	return
 }
+
+// GetRefreshTokens retrieves details for a user's refresh tokens.
+//
+// It allows pagination using the provided options. For more information on pagination, refer to:
+// https://pkg.go.dev/github.com/auth0/go-auth0/management#hdr-Page_Based_Pagination
+//
+// See: https://auth0.com/docs/api/management/v2#!/Users/get-refresh-tokens-for-user
+func (m *UserManager) GetRefreshTokens(ctx context.Context, userID string, opts ...RequestOption) (r *RefreshTokenList, err error) {
```
Comment (sergiught) [inline management/user.go:759]:
```suggestion
func (m *UserManager) ListRefreshTokens(ctx context.Context, userID string, opts ...RequestOption) (r *RefreshTokenList, err error) {
```

Should we call this `ListRefreshTokens` so it's similar to the pre-established convention throughout the SDK?

File: management/user_test.go [hunk 2]
```diff
@@ -483,9 +570,26 @@ func givenAUser(t *testing.T) *User {
...
+}
+
 func cleanupUser(t *testing.T, userID string) {
 	t.Helper()
 
 	err := api.User.Delete(context.Background(), userID)
 	require.NoError(t, err)
 }
+
+func skipE2E(t *testing.T) {
```
Comment (sergiught) [inline management/user_test.go:589]:
```suggestion
func skipTestIfRunningE2E(t *testing.T) {
```

Let's be a bit more descriptive as to what this does.

File: management/user_test.go [hunk 3]
```diff
@@ -483,9 +570,26 @@ func givenAUser(t *testing.T) *User {
...
+	t.Helper()
+
+	if !httpRecordingsEnabled {
+		t.Skip("Skipped as cannot be test in E2E scenario")
```
Comment (sergiught) [inline management/user_test.go:593]:
```suggestion
		t.Skip("Skipped due to inability of setting this up for an E2E scenario")
```

File: management/user_test.go [hunk 4]
```diff
@@ -444,6 +444,93 @@ func TestUserManager_Organizations(t *testing.T) {
...
+	require.NoError(t, err)
+	assert.Equal(t, expectedTokens, tokens.Tokens)
+	assert.Equal(t, "RefreshTokenID", tokens.Next)
+}
+
+// TestUserManager_DeleteRefreshTokens tests the DeleteRefreshTokens method of UserManager.
+// This E2E test is skipped because refresh tokens cannot be created without UI interaction.
+func TestUserManager_DeleteRefreshTokens(t *testing.T) {
+	configureHTTPTestRecordings(t)
+	skipE2E(t)
```
Comment (sergiught) [inline management/user_test.go:492]: Should skip be the first check we do?

File: management/user_test.go [hunk 5]
```diff
@@ -444,6 +444,93 @@ func TestUserManager_Organizations(t *testing.T) {
...
+
+	user := &User{ID: auth0.String("UserID")}
+	expectedToken1 := RefreshToken{
```
Comment (sergiught) [inline management/user_test.go:495]: Maybe it's worth mentioning through a comment here that the test recording was manually generated to match these details and if something changes here we'll need to modify manually the test recording

File: management/user_test.go [hunk 6]
```diff
@@ -483,9 +570,26 @@ func givenAUser(t *testing.T) *User {
 	return user
 }
 
+func givenRefreshTokens(t *testing.T) *RefreshTokenList {
```
Comment (sergiught) [inline management/user_test.go:573]: This doesn't really create a refresh token, just retrieves them from the User.

**Review by ewanharris — APPROVED** [review]

**Pull request merged**